react/features/toolbox/subscriber.ts

```diff
@@ -18,3 +24,85 @@ StateListenerRegistry.register(
         }
     }
 );
+
+
+const checkToolboxOverlap = (clientHeight: number, store: IStore) => {
+    let toolboxRect = document.querySelector('.toolbox-content-items')?.getBoundingClientRect();
+
+    if (!toolboxRect) {
+        return;
+    }
+    const tiles = document.querySelectorAll('span.videocontainer');
+
+    if (!tiles.length) {
+        return;
+    }
+
+    const toolboxHeight = 48 + 12; // height + padding
+    const bottomMargin = 16;
+
+    // Set top and bottom manually to avoid wrong coordinates
+    // caused by the hiding/ showing of the toolbox.
+    toolboxRect = {
+        ...toolboxRect,
+        top: clientHeight - toolboxHeight - bottomMargin,
+        bottom: clientHeight - bottomMargin,
+        left: toolboxRect.left,
+        right: toolboxRect.right
+    };
+    let isIntersecting = false;
+
+    const rows = store.getState()['features/filmstrip'].tileViewDimensions?.gridDimensions?.rows;
+    const noOfTilesToCheck = rows === 1 ? tiles.length : DEFAULT_MAX_COLUMNS - 1;
+
+    for (let i = 1; i < Math.max(noOfTilesToCheck, tiles.length); i++) {
+        const tile = tiles[tiles.length - i];
+        const indicatorsRect = tile?.querySelector('.bottom-indicators')?.getBoundingClientRect();
+
+        if (!indicatorsRect) {
+            // eslint-disable-next-line no-continue
+            continue;
+        }
+
+        if (!(
+            indicatorsRect.top > toolboxRect.bottom
+            || indicatorsRect.right < toolboxRect.left
+            || indicatorsRect.bottom < toolboxRect.top
+            || indicatorsRect.left > toolboxRect.right
+        )) {
```

Comment:
for readability:
```
        if (indicatorsRect.top <= toolboxRect.bottom
            && indicatorsRect.right <= toolboxRect.left
            && indicatorsRect.bottom <= toolboxRect.top
            && indicatorsRect.left <= toolboxRect.right
        ))
```

Reply:
doesn't this only check when tooblox completely cover the indicator?

Reply:
also, couldn't the IntersectionObserver be used?

Reply:
> also, couldn't the IntersectionObserver be used?

Not really. Because the toolbox can also be hidden or animating the show/ hide. If the toolbox is hidden when the subscriber function is called we'd have no intersection. To fix this we'd need to also listen for toolbox show/ hide, but that would case other problems and lots of unnecessary calculations

Reply:
> 

I applied de Morgan's law on your condition, should be <=> with what you have :P I think this is a reason why !(... )is not so recommended as it's confusing.

Reply:
> 

I applied de Morgan's law on your condition, should be <=> with what you have :P I think this is a reason why !(... )is not so recommended as it's confusing.

Reply:
Thanks for the tutorial but it's not necessary :P
Considering this code will become obsolete pretty much immediately, if you don't feel strongly about it I'll leave it as is